### PR TITLE
Create KitapYurdu.com.js

### PR DESCRIPTION
**Pull request by has-hus**
Translator for an online Turkish book store.

### CODE REVIEW

**Comment by adam3smith** [comment]
This looks quite neat, thanks. Could you add some tests as well as an (AGPL3) license?

File: KitapYurdu.com.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
+	"translatorID": "d3f35d5a-55da-4e07-be7d-b4d2a821279f",
+	"label": "KitapYurdu.com",
+	"creator": "Hasan Huseyin DER",
+	"target": "^http?://www\\.kitapyurdu\\.com/",
```
Comment (zuphilip) [inline KitapYurdu.com.js]: This should start with `"^https?://"` to work for `http` as well as `https`.

File: KitapYurdu.com.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
```
Comment (zuphilip) [inline KitapYurdu.com.js]: Activate here all browsers and platforms, normally they work as well.

File: KitapYurdu.com.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	
+	var edition = ZU.trimInternal(doc.querySelector('[itemprop=bookEdition]').textContent);
+	item.edition = edition.split('.')[0];
+	
+	var publisher = ZU.trimInternal(doc.querySelector('[itemprop=publisher]').textContent);
+	item.publisher = ZU.capitalizeTitle(publisher,true);
+	
+	item.date = ZU.trimInternal(doc.querySelector('[itemprop=datePublished]').textContent);
+	item.ISBN = ZU.trimInternal(doc.querySelector('[itemprop=isbn]').textContent);
+	item.numPages = ZU.trimInternal(doc.querySelector('[itemprop=numberOfPages]').textContent);
```
Comment (zuphilip) [inline KitapYurdu.com.js]: Did you try the Embedded Metadata translator as a starting point? It looks that there is some metadata available there.

**Comment by has-hus** [comment]
Thanks for the inputs.

@zuphilip I assume something's missing in the site's html, Embedded Metadata translator did not extract book info, just gave webpage...

**Comment by adam3smith** [comment]
Yeah, the site has almost no embedded metadata in the header, so the Zotero EM translator is of no help here. It does have schema.org microdata embedded, but we're not currently trying to get that in the embedded metadata translator, so I think this is the right approach.

File: KitapYurdu.com.js [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+		item.creators.push(ZU.cleanAuthor(creator, "translator"));
+	}
+	
+	var editors = ZU.xpath(doc, '//tr[contains(., "Editor")]|//tr[contains(., "Derleyici")]');
+	for (var i=0; i<editors.length; i++) {
+		var creator = cleanCreatorTitles(editors[i].textContent);
+		item.creators.push(ZU.cleanAuthor(creator, "editor"));
+	}
+	
+	var edition = ZU.trimInternal(doc.querySelector('[itemprop=bookEdition]').textContent);
```
Comment (adam3smith) [inline KitapYurdu.com.js]: I'm a bit worried this is fragile to the absence of an edition. `ZU.trimInternal` and `textContent` will both throw an error when used on null. I'd do this in two steps:
```
var edition = doc.querySelector('[itemprop=bookEdition]');
if (edition) edition = ZU.trimInternal(edition.textContent);
```
same for all the following fields. Or you could write a function that does this and returns either the trimemd text or `null`. Setting an item element to null is fine.

Comment (has-hus) [inline KitapYurdu.com.js]: @adam3smith Thanks. 

I tried to update them as you suggested.

**Comment by adam3smith** [comment]
great, thanks!

**Pull request merged**